tigera/operator
# Operator managed Prometheus CR

## Description

This changeset brings installation of Prometheus related custom
resources under the control of Tigera operator.

* Add a new MonitorSpec with no user configurable fields for this PR.
* Add Monitor controller.
* Add Monitor render.

After this change, Prometheus related CRs will be removed from
`custom-resource.yaml`. It will be replaced with a new Monitor CR.

Manual test steps:

1. Provision a k8s cluster and install Calico Enterprise.
2. Delete Prometheus CRs and verify Alertmanager and Prometheus pods are terminated.
3. Apply the new Monitor CRD.
4. Apply the new Monitor CR.
5. Apply the updated Tigera operator role.
6. Update tigera-operator deployment spec with the custom built image.
7. Watch operator logs and tigerastatus (all available / no degraded).
8. Verify Alertmanager and Prometheus pods are running.
9. Repeat the above steps for vanilla k8s on AWS and GCP and Openshift.



## For PR author

- [x] Tests for change.
- [x] If changing pkg/apis/, run `make gen-files`
- [x] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.
